Upstream-Status: Accepted

the patch is generated based on following commit of gcc official git tree:
http://gcc.gnu.org/git/?p=gcc.git;a=commitdiff;h=35e5dc9fa7f94c0cab5897abfd677034b584539c

Signed-off-by: Zhenhua Luo <b19537@freescale.com>
diff --git a/gcc/gengtype.c b/gcc/gengtype.c
index abf17f8..6c0ca4a 100644
--- a/gcc/gengtype.c
+++ b/gcc/gengtype.c
@@ -3594,14 +3594,13 @@ write_field_root (outf_p f, pair_p v, type_p type, const char *name,
 		  int has_length, struct fileloc *line, const char *if_marked,
 		  bool emit_pch, type_p field_type, const char *field_name)
 {
+  struct pair newv;
   /* If the field reference is relative to V, rather than to some
      subcomponent of V, we can mark any subarrays with a single stride.
      We're effectively treating the field as a global variable in its
      own right.  */
   if (v && type == v->type)
     {
-      struct pair newv;
-
       newv = *v;
       newv.type = field_type;
       newv.name = ACONCAT ((v->name, ".", field_name, NULL));
-- 
1.7.3.4
